File: backend/src/main/java/ipeps/pwd/wallet/payload/updatePayload/EmployeeUpdatePayload.java
package ipeps.pwd.wallet.payload.updatePayload;

import ipeps.pwd.wallet.entity.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class EmployeeUpdatePayload {
    private UUID employee_id;
    private String role;
    private boolean actif;
    private Account account;
    private Organization organization;
    private List<Address> addresses;
    private List<Salary> salaries;
    private List<Document> documents;
    private List<Wallet> wallets;
    private List<Schedule> schedules;
    private List<MessageAction> messageActions;
    private List<Message> messages;
}

File: frontend/src/app/modules/employee/model/payload/employee-update.payload.ts
import {PayloadInterface} from "@shared/model";
import {Organization} from "@organization/model";
import {Account} from "@account/model";

export interface EmployeeUpdatePayload extends PayloadInterface{
  employee_id: string;
  role: string;
  actif: boolean;
  organization: Organization;
  account: Account;
}
